app/src/main/java/org/jellyfin/mobile/player/PlayerActivity.kt

```diff
@@ -317,9 +318,30 @@ class PlayerActivity : AppCompatActivity() {
                 return true
             }
         })
+        // Handles scale/zoom gesture
+        val zoomGestureDetector = ScaleGestureDetector(this, object : ScaleGestureDetector.OnScaleGestureListener {
+            override fun onScaleBegin(detector: ScaleGestureDetector): Boolean = true
+
+            override fun onScale(detector: ScaleGestureDetector): Boolean {
+                val scaleFactor = detector.scaleFactor
+                if (abs(scaleFactor - 1f) > 0.01f) {
+                    playerView.resizeMode = if (scaleFactor > 1) AspectRatioFrameLayout.RESIZE_MODE_ZOOM else AspectRatioFrameLayout.RESIZE_MODE_FIT
+                }
+                return true
+            }
+
+            override fun onScaleEnd(detector: ScaleGestureDetector) {}
+        }).apply { isQuickScaleEnabled = false }
 
         playerView.setOnTouchListener { _, event ->
-            if (playerView.useController) gestureDetector.onTouchEvent(event) else unlockDetector.onTouchEvent(event)
+            if (playerView.useController) {
+                when (event.pointerCount) {
+                    1 -> gestureDetector.onTouchEvent(event)
+                    // Apparently the scale gesture detector doesn't enforce a maximum finger count.
+                    // If anyone scaled with more than 5 fingers, I'd be worried.
+                    in 2..5 -> zoomGestureDetector.onTouchEvent(event)
```

Comment:
I think we should only allow zooming with 2 fingers

Reply:
Maybe :sweat_smile: